vscode-extension/PSA_custom_Rules/Alias/avoidAlias.psm1

```diff
@@ -0,0 +1,92 @@
+        Import-Module $getAliasSpecFunctionFile
+
+        #get the alias mapping data
+        $aliasSpecFile = ".\PSA_custom_Rules\Alias\AliasSpec.json"
+        $AliasSpec = Get-AliasSpec -AliasPath $aliasSpecFile
+
+        # get the commandAst in the file
+        $foundCmdlets = Find-CmdletsInFile -rootAstNode $testAst
+    
+        #list of CorrectionExtents
+        $l = (new-object System.Collections.ObjectModel.Collection["Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic.CorrectionExtent"])
+
+        foreach ($cmdletReference in $foundCmdlets) {
+            if ($AliasSpec.cmdlet.psobject.properties.match($cmdletReference.CommandName).Count) {
+                [int]$startLineNumber = $cmdletReference.StartLine
+                [int]$endLineNumber = $cmdletReference.EndLine
+                [int]$startColumnNumber = $cmdletReference.StartColumn
+                [int]$endColumnNumber = $cmdletReference.EndPosition
+                [string]$correction = $AliasSpec.cmdlet.($cmdletReference.CommandName)
+                [string]$filePath = $cmdletReference.FullPath
+                [string]$optionalDescription = 'The alias can be changed to be formal name.'
+
+                $c = (new-object Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic.CorrectionExtent $startLineNumber, $endLineNumber, $startColumnNumber, $endColumnNumber, $correction, $filePath, $optionalDescription)
+                $l.Add($c)
+                
+            }
+
+            if ($AliasSpec.para_cmdlet.psobject.properties.match($cmdletReference.CommandName).Count){
+                foreach ($para in $cmdletReference.parameters){
+                    if ($AliasSpec.para_cmdlet.($cmdletReference.CommandName).psobject.properties.match($para.Name).Count){
+                        [int]$startLineNumber = $para.StartLine
+                        [int]$endLineNumber = $para.EndLine
+                        [int]$startColumnNumber = $para.StartColumn
+                        [int]$endColumnNumber = $para.EndPosition
+                        [string]$correction = $AliasSpec.para_cmdlet.($cmdletReference.CommandName).($para.Name)
+                        [string]$filePath = $para.FullPath
+                        [string]$optionalDescription = 'The alias can be changed to be formal name.'
+
+                        $c = (new-object Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic.CorrectionExtent $startLineNumber, $endLineNumber, $startColumnNumber, $endColumnNumber, $correction, $filePath, $optionalDescription)
+                        $l.Add($c)
+                    }
+                }
+            }
+        }
+        
+
+
+        $extent = $null
+        
+        #returned anaylse results
+        $dr = New-Object `
+            -Typename "Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic.DiagnosticRecord" `
+            -ArgumentList "This is help", $extent, $PSCmdlet.MyInvocation.InvocationName, Warning, "MyRuleSuppressionID", $l
```

Comment:
Is the first argument used? If not, please explicitly say it.
Also is MyRuleSuppressionID used?
```suggestion
            -ArgumentList "This arugment is not used.", $extent, $PSCmdlet.MyInvocation.InvocationName, Warning, "MyRuleSuppressionID", $l
```